sdk/storage/storage-queue/src/generated/lib/operations/service.ts

```diff
@@ -115,25 +115,53 @@ export class Service {
   /**
    * @param callback The callback
    */
-  listQueuesSegment(callback: msRest.ServiceCallback<Models.ListQueuesSegmentResponse>): void;
+  listQueuesSegment(callback: coreHttp.ServiceCallback<Models.ListQueuesSegmentResponse>): void;
   /**
    * @param options The optional parameters
    * @param callback The callback
    */
-  listQueuesSegment(options: Models.ServiceListQueuesSegmentOptionalParams, callback: msRest.ServiceCallback<Models.ListQueuesSegmentResponse>): void;
-  listQueuesSegment(options?: Models.ServiceListQueuesSegmentOptionalParams | msRest.ServiceCallback<Models.ListQueuesSegmentResponse>, callback?: msRest.ServiceCallback<Models.ListQueuesSegmentResponse>): Promise<Models.ServiceListQueuesSegmentResponse> {
+  listQueuesSegment(options: Models.ServiceListQueuesSegmentOptionalParams, callback: coreHttp.ServiceCallback<Models.ListQueuesSegmentResponse>): void;
+  listQueuesSegment(options?: Models.ServiceListQueuesSegmentOptionalParams | coreHttp.ServiceCallback<Models.ListQueuesSegmentResponse>, callback?: coreHttp.ServiceCallback<Models.ListQueuesSegmentResponse>): Promise<Models.ServiceListQueuesSegmentResponse> {
     return this.client.sendOperationRequest(
       {
         options
       },
       listQueuesSegmentOperationSpec,
       callback) as Promise<Models.ServiceListQueuesSegmentResponse>;
   }
+
+  /**
+   * The List Queues Segment operation returns a list of the queues under the specified account
+   * @param nextPageLink The NextLink from the previous successful call to List operation.
+   * @param [options] The optional parameters
+   * @returns Promise<Models.ServiceListQueuesSegmentResponse>
+   */
+  listQueuesSegmentNext(nextPageLink: string, options?: Models.ServiceListQueuesSegmentNextOptionalParams): Promise<Models.ServiceListQueuesSegmentResponse>;
```

Comment:
Previously there's only one `listQueueSegment()` method.  Now we have both `listQueuesSegment()` and `listQueuesSegmentNext()` I wonder whether we there are any changes that we need to make in how we list queues.

Reply:
I don't think any changes are necessary.  The reason why these showed up now is because I updated `autorest.typescript` to include methods for operations with results that support paging when the `--azure-arm` parameter *isn't* used.  These methods can be used when paging is desired but I think they can be ignored otherwise.

Reply:
We do need paging: (https://github.com/Azure/azure-sdk-for-js/blob/feature/storage/sdk/storage/storage-queue/src/QueueServiceClient.ts#L289-L298)

Reply:
Looks like `listQueuesSegment()` didn't change so it should still work. I guess `listQueuesSegmentNext()` is new addition that takes the marker - `nextPageLink` explicitly.

Reply:
Yep, that's correct.  Interesting that results can be paged without these additions, I wonder why the `keyvault-*` libraries needed the explicit `*Next()` methods...